src/libraries/System.Text.Json/gen/Helpers/KnownTypeSymbols.cs

```diff
@@ -141,6 +141,18 @@ public KnownTypeSymbols(Compilation compilation)
 
         private Option<IArrayTypeSymbol?> _ByteArrayType;
 
+        public INamedTypeSymbol? MemoryByteType => _MemoryByteType.HasValue
```

Comment:
I think querying the `_MemoryByteType` field directly means that the presence of a value will depend on whether that field has been populated by a previous call. I would recommend just adding a symbol for `Byte` and calling the `Construct` method from the parser directly.

Reply:
How is that not a problem for the existing `_ByteArrayType`?
https://github.com/dotnet/runtime/blob/dedaf46154a8938848fc9a7e0bd6e5ccebe7809c/src/libraries/System.Text.Json/gen/Helpers/KnownTypeSymbols.cs#L138

Reply:
I was reviewing this over the phone this morning and actually misread what it was doing. You're right that this is fine, apologies for the confusion.

Reply:
> will depend on whether that field has been populated by a previous call

Given that this is currently working fine, in which case there isn't a "previous call" that could cause trouble?